### DISCUSSION TITLE
觉得Draggable的功能其实挺适用的，在Modal或者Tree里面

### DISCUSSION THREAD

**Discussion by JackDan9** [DouyinFE/semi-design#1705] [Ideas]
是否有考虑这部分功能做成组件

**Comment by pointhalo** [comment]
没看明白。是希望 Modal 可以支持在屏幕内拖拽移动？还是指Modal内部的元素可以拖拽

**Reply by JackDan9** [reply]
Modal就是Modal在屏幕内的拖拽移动

**Reply by JackDan9** [reply]
但是Tree的就是节点间的移动交换，只是我觉得draggable其实可以去实现一下

**Reply by JackDan9** [reply]
我实现一个效果放到codesandbox里面，放出来，但是我觉得可以做成组件或者组件中的一类属性

**Reply by pointhalo** [reply]
1. 针对整个 Component 提供拖拽的功能，像 [dnd-kit](https://docs.dndkit.com/) 这类社区中的包已经做得非常成熟了。简单组合使用就能实现，我们没有计划再内置重新实现一套
2.  Tree 的拖拽我还是没看明白你说的具体是什么。目前节点交换已经是有了的功能。https://semi.design/zh-CN/navigation/tree#%E5%8F%AF%E6%8B%96%E6%8B%BD%E7%9A%84Tree

**Reply by JackDan9** [reply]
我写一下效果，dnd-kit是实现了，但是其实也是需要成本投入的，我的想法是Modal可以支持
